google/cloud/bigquery/job/query.py

```diff
@@ -1515,11 +1529,25 @@ def result(  # type: ignore  # (incompatible with supertype)
                 # actually correspond to a finished query job.
             )
 
+        # Setting max_results should be equivalent to setting page_size with
+        # regards to allowing the user to tune how many results to download
+        # while we wait for the query to finish. See internal issue:
+        # 344008814.
+        if page_size is None and max_results is not None:
```

Comment:
Just for my education, why are setting max_results and page_size equivalent here?

Reply:
Imagine the user has selected a value for `max_results`. That means we should not download more than that number of rows, in which case it's important to set the page size.

Also, it means the user is expecting to download the results and do something with them, which means it wouldn't be a waste to ask for the first page of results while we wait for the query to finish.